lib/Service/TextBlockService.php

```diff
@@ -0,0 +1,190 @@
+		$this->groupManager = $groupManager;
+	}
+
+	/**
+	 * @param string $userId
+	 * @return TextBlock[]
+	 */
+	public function findAll(string $userId): array {
+		return $this->textBlockMapper->findAll($userId);
+	}
+
+	/**
+	 * @param string $userId
+	 * @return TextBlock[]
+	 * @throws DoesNotExistException
+	 */
+	public function findAllSharedWithMe(string $userId): array {
+		$user = $this->userManager->get($userId);
+		if ($user === null) {
+			throw new DoesNotExistException('User does not exist');
+		}
+		$groups = $this->groupManager->getUserGroupIds($user);
+		return $this->textBlockMapper->findSharedWithMe($userId, $groups);
+	}
+	/**
+	 * @param int $textBlockId
+	 * @param string $userId
+	 * @return TextBlock|null
+	 */
+	public function find(int $textBlockId, string $userId): ?TextBlock {
+		return $this->textBlockMapper->find($textBlockId, $userId);
+	}
+
+	/**
+	 * @param string $userId
+	 * @param string $title
+	 * @param string $content
+	 * @return TextBlock
+	 */
+	public function create(string $userId, string $title, string $content): TextBlock {
+		$textBlock = new TextBlock();
+		$textBlock->setContent($content);
+		$textBlock->setOwner($userId);
+		$textBlock->setTitle($title);
+		$html = new Html2Text($content, ['do_links' => 'none','alt_image' => 'hide']);
+		$preview = trim($html->getText());
+		$textBlock->setPreview(substr($preview, 0, 50));
+		return $this->textBlockMapper->insert($textBlock);
+	}
+
+	/**
+	 * @param int $textBlockId
+	 * @param string $userId
+	 * @param string $title
+	 * @param string $content
+	 * @return TextBlock
+	 */
+	public function update(int $textBlockId, string $userId, string $title, string $content): TextBlock {
+		$textBlock = $this->textBlockMapper->find($textBlockId, $userId);
+		if ($textBlock === null) {
+			throw new DoesNotExistException('TextBlock does not exist');
+		}
+		$textBlock->setContent($content);
+		$textBlock->setTitle($title);
+		$html = new Html2Text($content, ['do_links' => 'none','alt_image' => 'hide']);
+		$preview = trim($html->getText());
+		$textBlock->setPreview(substr($preview, 0, 300));
+		return $this->textBlockMapper->update($textBlock);
+	}
+
+	/**
+	 * @param int $textBlockId
+	 * @param string $userId
+	 * @throws DoesNotExistException
+	 */
+	public function delete(int $textBlockId, string $userId): void {
+		$textBlock = $this->textBlockMapper->find($textBlockId, $userId);
+		if ($textBlock === null) {
+			throw new DoesNotExistException('TextBlock does not exist');
+		}
+		$this->textBlockMapper->delete($textBlock);
+		$this->textBlockShareMapper->deleteByTextBlockId($textBlockId);
+	}
+
+
+	/**
+	 * @param int $textBlockId
+	 * @param string $shareWith
+	 * @throws DoesNotExistException
+	 * @throws NotPermittedException
+	 */
+	public function share(int $textBlockId, string $shareWith): void {
+
+		$sharee = $this->userManager->get($shareWith);
+		if ($sharee === null) {
+			throw new DoesNotExistException('Sharee does not exist');
```

Comment:
Those null checks for the user are quite frequent in this change, so it might be worth introducing a UserNotFoundException. For example, like polls and files_reminders: https://github.com/nextcloud/server/blob/master/apps/files_reminders/lib/Exception/UserNotFoundException.php